internal/engine/wazevo/engine.go

```diff
@@ -841,3 +894,14 @@ func (cm *compiledModule) getSourceOffset(pc uintptr) uint64 {
 	}
 	return cm.sourceMap.wasmBinaryOffsets[index]
 }
+
+func sequence(size int) <-chan int {
```

Comment:
probably more efficient to just use atomics and a yield function. but this is a microoptimization.

Reply:
yeah, rewrite this using a mutex and and a argument of a shuffle list of indices.

Reply:
Ncruces and I tried removing the channel and using an atomic and didn't have any noticeable performance gains.